src/traces/parcoords/defaults.js

```diff
@@ -17,14 +17,20 @@ var handleDomainDefaults = require('../../plots/domain').defaults;
 
 function handleLineDefaults(traceIn, traceOut, defaultColor, layout, coerce) {
 
-    coerce('line.color', defaultColor);
-
-    if(hasColorscale(traceIn, 'line') && Lib.isArray(traceIn.line.color)) {
-        coerce('line.colorscale');
-        colorscaleDefaults(traceIn, traceOut, layout, coerce, {prefix: 'line.', cLetter: 'c'});
-    }
-    else {
-        coerce('line.color', defaultColor);
+    var lineColor = coerce('line.color', defaultColor);
+
+    if(hasColorscale(traceIn, 'line') && Lib.isArray(lineColor)) {
+        if(lineColor.length) {
+            coerce('line.colorscale');
+            colorscaleDefaults(traceIn, traceOut, layout, coerce, {prefix: 'line.', cLetter: 'c'});
+            // TODO: I think it would be better to keep showing lines beyond the last line color
+            // but I'm not sure what color to give these lines - probably black or white
+            // depending on the background color?
```

Comment:
I agree here.

Currently, scatter traces with `marker.color` arrays shorter than the coordinates get black markers - regardless of `plot_bgcolor`. Adding some logic around the background color would be even better.

This probably deserves a new issue.

Reply:
Great -> #2405